pyomo/contrib/solver/base.py

```diff
@@ -418,9 +418,14 @@ def _map_results(self, model, results):
         ]
         legacy_soln.status = legacy_solution_status_map[results.solution_status]
         legacy_results.solver.termination_message = str(results.termination_condition)
-        obj = get_objective(model)
-        if len(list(obj)) > 0:
+        legacy_results.problem.number_of_constraints = model.nconstraints()
+        legacy_results.problem.number_of_variables = model.nvariables()
+        if model.nobjectives() == 0:
+            legacy_results.problem.number_of_objectives = 0
+        else:
+            obj = get_objective(model)
```

Comment:
This function does one of three things:
1. returns `None` if there is not an active objective
2. raises an error if there are multiple active objectives
3. returns the objective if there is exactly one objective

As a result, you will not be able to do `len(obj)` below.

Reply:
So I should just do `model.nobjectives()` is how I interpret that. Accurate?

Reply:
Yes. Then you just need to grab the objective to get the sense.

Reply:
Updated!

Reply:
yes

Reply:
So this patch will raise an `AttributeError` for multiobjective NL files (which is in theory supported by the ASL / Pyomo API), in line 428 `None.sense`

Reply:
What is the right way to get `sense` for multiple objectives?

Reply:
A single `sense` is meaningless for multiple objectives (it should be a vector).  But the old interface never really handled MO problems well.  That said, my concern is that this is probably not the right place for a MO problem to end up with an error.